src/sounds.cpp

```diff
@@ -481,6 +482,16 @@ void sounds::process_sounds()
                 critter.hear_sound( source, vol, dist, this_centroid.provocative );
             }
         }
+        // Trigger sound-triggered traps
+        for( const trap *trapType : trap::get_sound_triggered_traps() ) {
+            for( tripoint tp : get_map().trap_locations( trapType->id ) ) {
+                const int dist = sound_distance( source, tp );
+                trap tr = get_map().tr_at( tp );
```

Comment:
No need to copy these.
```suggestion
            for( const tripoint &tp : get_map().trap_locations( trapType->id ) ) {
                const int dist = sound_distance( source, tp );
                const trap &tr = get_map().tr_at( tp );
```